src/lib.rs

```diff
@@ -4,11 +4,18 @@ use ispc::WeightCollection;
 
 mod ispc;
 
-pub trait ImagePixelFormat {
-    fn num_channel_in_memory(&self) -> usize;
-    fn channel_size_in_bytes(&self) -> usize;
-
-    fn pixel_size(&self) -> usize {
+pub trait ImagePixelFormat: Copy {
+    /// Returns the number of channels that an image of this format would have in memory.
+    /// For example, while a normal map of format [`NormalMapFormat::R8g8TangentSpaceReconstructedZ`] would still have 3 channels when sampled,
+    /// in memory it will have 2 channels.
+    fn num_channel_in_memory(self) -> usize;
+
+    /// Returns the size of a single value channel, in bytes.
+    fn channel_size_in_bytes(self) -> usize;
+
+    /// Returns the size in bytes of a single pixel.
+    /// Generally this will be equal to [channel_size_in_bytes()][`ImagePixelFormat::channel_size_in_bytes()`] * [num_channel_in_memory()][`ImagePixelFormat::num_channel_in_memory()`].
```

Comment:
Wrong way around, I think this won't link anywhere now:

```suggestion
    /// Generally this will be equal to [`channel_size_in_bytes()`][Self::channel_size_in_bytes()] `*` [`num_channel_in_memory()`][Self::num_channel_in_memory()].
```